src/app/shared/utils/utils.ts

```diff
@@ -437,7 +437,15 @@ export function parseAnswerList(answerList: any) {
       return parseAnswerListItem(item);
     }
   );
-  return answerListItems;
+  // Remove any items from the list which only have a value for "name",
+  // e.g. "image" and "text" are undefined because the list has been generated within a template
+  const filteredAnswerListItems = answerListItems.filter((item: IAnswerListItem) => {
+    for (let [key, value] of Object.entries(item)) {
+      if (key !== "name" && value !== undefined) return true;
```

Comment:
nit(blocking)
This logic seems a bit confusing to me, we're running a filter operation but then have an inner for loop which will escape before completion. Might make more sense as 2-step operation, e.g more like
```ts
const hadItemData = Object.entries(item).find(([key,value])=>(key !== "name" && value !== undefined)
return hadItemData  ? true : false
```

That being considered, I'm not sure if this makes total sense to implement - surely it's better to let broken images through so that an author can be aware of an authoring mistake instead of just hiding the data away? I'd probably lean more towards providing an authoring option to add a filter function to the specific component, e.g. `filter_no_image:true` or ensuring support for runtime filtering, e.g. `@data.some_list.filter(item=>item.image!==undefined)`

Also our type definitions don't suggest either text or image are required, and we'd still end up with broken images in cases where text is supplied without image (but image expected) with the above filter.

Reply:
Your suggested logic is definitely clearer, I have refactored accordingly. 

In terms of allowing through broken images, a broken image link will still be allowed through with the suggested implementation – an invalid path to an image will still display a broken image link. The `image` will only be filtered out if the value is `undefined` – which admittedly could be due to an authoring error that the author should be made aware of, but it's not clear to me that displaying a broken image would be necessary in such cases. 

Either way, returning to the issue referenced in this PR, we want a way to author an answer list in a generic way where the source data may or may not contain image paths, and in the case where an item does not reference an image, to display no image (and not a broken image link). Filtering on the `undefined` values as suggested seems like a satisfactory solution, but I may be missing some associated problems, and there may be other approaches. 

Ultimately, I think the best solution may be a way of defining arbitrary data structures, or at least answer lists, directly from data lists, handled at parse-time, in a way that can handle lists of multiple length and containing partial values (as in [WIP RFC for a universal data list parser](https://docs.google.com/document/d/1XMKCu5ohiBXoBB1k5W5Kz7Bha8QSpkxn/edit#heading=h.gjdgxs)). But I think we want an acceptable solution sooner in order to help with the current authoring requirements.